### ISSUE TITLE
Split-CTM 2x2 gives different dense vs symmetric environments when chi_I << chi (random tensors)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#766]
Split-CTM on the `2x2` recipe returns a **different environment for `DenseTensor` vs `SymmetricTensor`** when the interlayer bond is aggressively truncated (`chi_I` well below `chi`), on a *random* site tensor. The charges are all trivial (a single block), so the two representations describe literally the same linear algebra and must agree.

## Measurement

D=2, d=2 random site (`jax.random.PRNGKey(42)`, normalized), all-zero U(1) charges, `chi=8`, `max_iter=50`:

| path | `chi_I` | E_dense | E_sym | gap |
|---|---|---|---|---|
| split `2x2` | 2 | -1.0512381821319083 | 0.8148446136370731 | 1.87 |
| split `2x2` | **4** | 0.29249418211883316 | **-14.316377305269969** | **1.46e+1** |
| split `2x2` | 6 | 0.4242920184056182 | 0.4242920184021381 | 3.5e-12 |
| split `2x2` | 8 (=chi) | 0.40885152472714914 | 0.408851524727149 | 1.7e-16 |
| split `2x2` | 12 | 0.39764097792599784 | 0.3976409779258555 | 1.4e-13 |
| split `1x1` | 4 and 8 | 0.04095962443784354 | 0.04095962443784193 | 1.6e-15 |
| fused `2x2` | — | 0.3853044482235 | 0.38530444822349996 | 5.6e-17 |

All 12 environment tensors differ (`maxdiff` 0.94 – 1.98), with matching shapes. The corner spectra at `chi_I=4`:

```
dense: 1.0, 0.603199, 0.304103, 0.184718, 0.137730, 0.043360, 0.015713, 0.000245
sym:   1.0, 0.752803, 0.580024, 0.416343, 0.390485, 0.075059, 0.027095, 0.008506
```

At `chi_I=chi` they are identical.

## Not single-site-specific, and not new

The 2-site checkerboard path (`ctm_split_tensor_2site(A, A, chi=8, chi_I=4)`) reproduces the gap to the digit — `14.608871487388804` vs `14.608871487388802` for the single-site path — so this lives in the shared `_split_ctm_sweep_multisite` 2x2 machinery, not in the 1-site neighbour-map delegation added by #746.

It was previously **masked**, not absent: on the `1x1` recipe the environment collapses to rank-1 corners (#726), so the interlayer bond has nothing to truncate and `chi_I` is inert — that path returns a bit-identical energy at `chi_I=4` and `chi_I=8`. Routing the single-site split path through `2x2` (#746) is what made it observable.

## Probably an ambiguous truncated subspace, not a correctness bug

Two things argue this is an under-resolved interlayer bond rather than a defect in the symmetric path:

1. **Dense is garbage there too.** Dense at `chi_I=4` (0.2925) is 0.11 away from its own converged `chi_I>=8` value (0.4089), and `chi_I=2` gives -1.05. Neither representation produces a meaningful energy in that regime; they produce *differently* meaningless ones.
2. **A physical state is clean at every `chi_I`.** On the D=2 sublattice-rotated Heisenberg simple-update state, dense and symmetric agree at `chi_I` = 2, 4, 6, 8, 12, 16 (gaps <= 1e-10), with the energy converging monotonically 0.4990414544 -> 0.4991253869.

So the retained interlayer subspace is genuinely ambiguous for a random tensor under hard truncation, and the dense and block-sparse SVDs resolve the tie differently. Production uses `chi_I = chi` on physical states and is unaffected.

## Why it is still worth tracking

The failure is **silent**: `-14.3` for a Heisenberg bond is unphysical by an order of magnitude, and nothing raises or warns. Candidate responses, in rough order of appeal:

- Warn (or refuse) when `chi_I` is small enough relative to the interlayer spectrum that the retained subspace is ambiguous — i.e. when the discarded interlayer weight is not small.
- Make the interlayer truncation tie-break deterministic across the dense and block-sparse SVD backends, so at least the two paths fail identically.
- Document `chi_I < chi` as unsupported for anything but bisection.

## Not established

- Whether the ambiguity is strictly degeneracy at the truncation boundary, or something sharper.
- Whether non-trivial (non-zero) charge sectors behave the same; all measurements above use trivial charges, which is the strongest form of the claim (one block) but not the general one.
- Whether the fermionic/graded split path shares it.

Found while landing #746. `tests/test_split_ctm_tensor.py::TestSplitCTMSymmetric::test_symmetric_energy_matches_dense` ran `chi_I=4, chi=8` and started failing; it now runs the production `chi_I=chi` and points here.

Related: #726, #746, #723.

> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.
